src/cloudwatch-mcp-server/awslabs/cloudwatch_mcp_server/cloudwatch_alarms/tools.py

```diff
@@ -0,0 +1,656 @@
+            if end_time is None or not isinstance(end_time, str):
+                end_time = datetime.now()
+            else:
+                end_time = datetime.fromisoformat(end_time.replace('Z', '+00:00'))
+            
+            if start_time is None or not isinstance(start_time, str):
+                start_time = end_time - timedelta(days=1)
+            else:
+                start_time = datetime.fromisoformat(start_time.replace('Z', '+00:00'))
+            
+            # Set default history item type
+            if not history_item_type:
+                history_item_type = 'StateUpdate'
+            
+            logger.info(f'Fetching alarm history for {alarm_name}')
+            logger.info(f'Time range: {start_time} to {end_time}')
+            
+            # Make the basic API call to GetAlarmHistory
+            params = {
+                'AlarmName': alarm_name,
+                'StartDate': start_time,
+                'EndDate': end_time,
+                'HistoryItemType': history_item_type,
+                'MaxRecords': max_items
+            }
+            
+            response = cloudwatch_client.describe_alarm_history(**params)
+            
+            # Transform history items with enhanced parsing
+            history_items = []
+            for item in response.get('AlarmHistoryItems', []):
+                history_item = self._transform_history_item(item)
+                history_items.append(history_item)
+            
+            # Get detailed alarm information
+            alarm_details = await self._get_alarm_details(cloudwatch_client, alarm_name)
+            
+            # Handle composite alarms if requested
+            if include_component_alarms and alarm_details.alarm_type == "CompositeAlarm":
+                return await self._handle_composite_alarm(cloudwatch_client, alarm_details)
+            
+            # Generate time range suggestions based on alarm history and configuration
+            time_range_suggestions = self._generate_time_range_suggestions(history_items, alarm_details)
+            
+            # Handle pagination
+            has_more_results = 'NextToken' in response
+            next_token = response.get('NextToken')
+            
+            # Create basic response
+            message = None
+            if len(history_items) == 0:
+                message = f"No alarm history found for {alarm_name} in the specified time range"
+            elif has_more_results:
+                message = f"Found {len(history_items)} history items with more results available"
+            
+            logger.info(f'Found {len(history_items)} alarm history items')
+            
+            return AlarmHistoryResponse(
+                alarm_details=alarm_details,
+                history_items=history_items,
+                time_range_suggestions=time_range_suggestions,
+                has_more_results=has_more_results,
+                next_token=next_token,
```

Comment:
What's the point in returning a next_token if the input args of this tool don't accept it? We should remove it.